internal/datacoord/meta.go

```diff
@@ -776,10 +774,15 @@ func (m *meta) CompleteMergeCompaction(compactionLogs []*datapb.CompactionSegmen
 
 	modSegments := make([]*datapb.SegmentInfo, 0, len(segments))
 	for _, s := range segments {
```

Comment:
what happens if this is an old segment with binlog path there.
Think about compatibility

Reply:
There is no compatibility issue because these old segments have already been marked SegmentState_Dropped state, and it has been converted to the new segment that excludes binlogs before reloading into meta cache.